DSCResources/MSFT_NetworkTeam/MSFT_NetworkTeam.psm1

```diff
@@ -0,0 +1,243 @@
+                if ($membersToRemove)
+                {
+                    Write-Verbose ($localizedData.removingMembers -f ($membersToRemove -join ','))
+                    $null = Remove-NetLbfoTeamMember -Name $membersToRemove -Team $name -ErrorAction Stop -Confirm:$false
+                }
+
+                $membersToAdd = ($netTeamMembers | Where-Object {$_.SideIndicator -eq '<='}).InputObject
+                if ($membersToAdd)
+                {
+                    Write-Verbose ($localizedData.addingMembers -f ($membersToAdd -join ','))
+                    $null = Add-NetLbfoTeamMember -Name $membersToAdd -Team $name -ErrorAction Stop -Confirm:$false
+                }
+            }
+            
+        } 
+        else 
+        {
+            Write-Verbose ($localizedData.createTeam -f $name)
+            $null = New-NetLbfoTeam -Name $name -TeamMembers $teamMembers -TeamingMode $teamingMode -LoadBalancingAlgorithm $loadBalancingAlgorithm -ErrorAction Stop -Confirm:$false
+            $timeout = 0
+            While ((Get-NetLbfoTeam -Name $name).status -ne 'Up')
+            {
+                Write-Verbose $localizedData.waitingForTeam
+                if ($timeout -ge 120)
+                {
+                    throw $localizedData.failedToCreateTeam   
+                }
+                Start-Sleep -Seconds 2
+                $timeout += 2
+            }
+
+            if ((Get-NetLbfoTeam -Name $name).status -eq 'Up')
+            {
+                Write-Verbose $localizedData.createdNetTeam
+            }
+        }
+    }
+    else
+    {
+        Write-Verbose ($localizedData.removeTeam -f $name)
+        $null = Remove-NetLbfoTeam -Name $name -ErrorAction Stop -Confirm:$false
+    }
+}
+
+Function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    Param
+    (
+        [Parameter(Mandatory)]
+        [string]$name,
+
+        [Parameter(Mandatory)]
+        [String[]]$teamMembers,
+    
+        [Parameter()]
+        [ValidateSet("SwitchIndependent", "LACP", "Static")]
+        [String]$teamingMode = "SwitchIndependent",
+
+        [Parameter()]
+        [ValidateSet("Dynamic", "HyperVPort", "IPAddresses", "MacAddresses", "TransportPorts")]
+        [String]$loadBalancingAlgorithm = "HyperVPort",
+
+        [ValidateSet('Present', 'Absent')]
+        [String]$ensure = 'Present'
+    )
+    
+    Write-Verbose ($localizedData.GetTeamInfo -f $name)
+    $networkTeam = Get-NetLBFOTeam -Name $name -ErrorAction SilentlyContinue
+    
+    if ($ensure -eq 'Present')
+    {
+        if ($networkTeam)
+        {
+            Write-Verbose ($localizedData.foundTeam -f $name)
+            if (
+                ($networkTeam.LoadBalancingAlgorithm -eq $loadBalancingAlgorithm) -and 
+                ($networkTeam.teamingMode -eq $teamingMode) -and 
+                ($null -eq (Compare-Object -ReferenceObject $teamMembers -DifferenceObject $networkTeam.Members))
```

Comment:
This will probably have the same problem as L58 with the difference in Members and TeamMembers